### PR TITLE
AEROGEAR-2451 - keycloak config implementation

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-2451

## Description

Keycloak configuration implementation

## Progress

- [x] `KeycloakConfig` class
- [x] Unit testing
- [x] Docs

### CODE REVIEW

File: Auth/Auth.Tests/Auth.Tests.csproj [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+  <ItemGroup>
+    <Folder Include="Resources\" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Remove="Resources\keycloak-config.json" />
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\keycloak-config.json">
+      <LogicalName>AeroGear.Mobile.Auth.Tests.Resources.keycloak-config.json</LogicalName>
+    </EmbeddedResource>
```
Comment (wei-lee) [inline Auth/Auth.Tests/Auth.Tests.csproj]: For testing, I don't this file should be in the Auth.Tests project. But it should be in the Auth.Tests.Android or Auth.Tests.iOS project instead. This file should never live inside a shared project

File: Auth/Auth.Tests/Resources/keycloak-config.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿{
```
Comment (wei-lee) [inline Auth/Auth.Tests/Resources/keycloak-config.json]: can we rename this to `mobile-services.json`

File: Auth/Auth.Tests/Auth.Tests.csproj (hunk 1 above)
Comment (wei-lee) [reply Auth/Auth.Tests/Auth.Tests.csproj]: Actually, never mind. Looks like it is easier to test for the Config class this way.

**Comment by rachael-oregan** [comment]
@wei-lee I've just included the keycloak changes here in a new PR -> https://github.com/aerogear/aerogear-xamarin-sdk/pull/18

Even when I just built the Auth and Auth.Tests folder, git was still showing changes in another file in the example app. No idea why

**Pull request closed without merging**